CMakeLists.txt

```diff
@@ -79,6 +86,11 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL iOS)
 	set(suffix_platform _ios)
 elseif(CMAKE_SYSTEM_NAME STREQUAL Android)
 	set(suffix_platform _android)
+elseif(CMAKE_SYSTEM_NAME STREQUAL Emscripten)
+	set(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-sSIDE_MODULE=1")
+	set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "-sSIDE_MODULE=1")
```

Comment:
I'm not seeing anything about these in the reference documentation. What's the purpose of this exactly?

Reply:
Yeah, I didn't either. My understanding is that it allows you to override the flag that instructs the compiler to produce a dynamically linked/shared library.

I nabbed this from here: https://github.com/emscripten-core/emscripten/pull/16281/files

Emscripten's support for dlink is still new and a bit shaky, so they don't have support rolled into their toolchain yet. This can probably all go away once that PR lands. (though it seems like it has stalled, so...)

I'll add a comment to clarify.

Reply:
Judging by how I've seen other `*_CREATE_*` flags used in CMake, I'm assuming this is just used by toolchains to add to the default set of `C[XX]_FLAGS`, so I would imagine this `-sSIDE_MODULE=1` could also just be added as part of the `target_compile_options(...)`.

Reply:
Likely so, though in this case I personally think it makes more sense to keep it isolated, as it really isn't a compile option that is meant to be changed. It would normally be buried away with other low(er) level platform-specific implementation details, and _hopefully_ won't be needed long-term.

It's kinda like a non-user serviceable part in a "fancy" IoT enabled coffee maker. 😼

Reply:
Are you sure this is actually doing anything to begin with? I tried setting it myself just now, both before and after `project(...)` as well as in the toolchain file for Android, and none of them seemed to result in this flag actually being added to any of the Ninja files.

Reply:
Yep! It warns at the final link:
```
em++: warning: linking a library with `-shared` will emit a static object file.  This is a form of emulation to support existing build systems.  If you want to build a runtime shared library use the SIDE_MODULE setting. [-Wemcc]
```

And then this when I try to run:
```
need the dylink section to be first
displayFailureNotice @ tmp_js_export.html:140
Promise.then (async)
(anonymous) @ tmp_js_export.html:186
(anonymous) @ tmp_js_export.html:190
```

The resulting output is ~55 MiB vs ~4.5 MiB too, which I didn't really expect. :shrug:

Reply:
Fair enough.

Either way, I'm not all that convinced about using these `*_CREATE_*` variables. This `-sSIDE_MODULE` flag is documented quite clearly [here](https://emscripten.org/docs/compiling/Dynamic-Linking.html#load-time-dynamic-linking) so I don't see much of a point trying to isolate it. None of the flags added through `target_compile_options(...)` are "user-serviceable" in the first place, as far as I know, unlike modifications to the global `C[XX]_FLAGS`.

Reply:
Oh wait, I had completely glossed over the fact that this was just temporary until they merge that PR on their end.

Nevermind!